sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2766,7 +2767,8 @@ class DataFrameSuite extends QueryTest
         // The data set has 2 partitions, so Spark will write at least 2 json files.
         // Use a non-splittable compression (gzip), to make sure the json scan RDD has at least 2
         // partitions.
-        .write.partitionBy("p").option("compression", "gzip").json(path.getCanonicalPath)
+        .write.partitionBy("p")
+        .option("compression", GZIP.lowerCaseName()).json(path.getCanonicalPath)
```

Comment:
it's recommended to eliminate the `()` on calling Java no-arg method which has no side-effects
```suggestion
        .option("compression", GZIP.lowerCaseName).json(path.getCanonicalPath)
```

Reply:
Because there are already a lot of Java no-arg method called with ().
Let's update them if we really have the strong requirements.

Reply:
@pan3793 Could these cases possibly manifest as compilation warnings? Or are they merely suggestions for best practices?

Reply:
It's a best practice, no compile warnings, because the compiler has no idea will this method have side effects.

Suggestion about [uniform access principle](https://en.wikipedia.org/wiki/Uniform_access_principle) from [Databricks Scala Guide](https://github.com/databricks/scala-style-guide#parentheses) and IntelliJ IDEA

> Reports calls to Java accessor methods with empty argument clauses.
>
> Methods that follow the JavaBean naming contract for accessors are expected to have no side effects. The recommended convention is to use a parameterless method whenever there are no parameters and the method have no side effect. This convention promotes the [uniform access principle](https://en.wikipedia.org/wiki/Uniform_access_principle), which says that the client code should not be affected by the decision to implement an attribute as a field or method.
> The problem is that Java does not implement the uniform access principle. To bridge that gap, Scala allows you to leave off the empty parentheses on an invocation of a Java method that takes no arguments.
> 
> The quick-fix removes the empty argument clause.
> Example:
> ```
> "test".getClass()
> ```
> After the quick-fix is applied:
> ```
> "test".getClass
> ```

Reply:
> Because there are already a lot of Java no-arg method called with ().

@beliefer the key point is calling of "no-arg method **without side effects**"

Reply:
According to the discussion offline between @cloud-fan and me. This change no need here.
Spark will add the compilation warnings if Java no-arg method called without ().